core/src/main/java/org/elasticsearch/action/admin/indices/upgrade/post/TransportUpgradeAction.java

```diff
@@ -130,8 +151,8 @@ protected ShardUpgradeResponse newShardResponse() {
     @Override
     protected ShardUpgradeResponse shardOperation(ShardUpgradeRequest request) {
         IndexShard indexShard = indicesService.indexServiceSafe(request.shardId().getIndex()).shardSafe(request.shardId().id());
-        org.apache.lucene.util.Version version = indexShard.upgrade(request.upgradeRequest());
-        return new ShardUpgradeResponse(request.shardId(), indexShard.routingEntry().primary(), version);
+        org.apache.lucene.util.Version luceneVersion = indexShard.upgrade(request.upgradeRequest());
+        return new ShardUpgradeResponse(request.shardId(), indexShard.routingEntry().primary(), Version.CURRENT, luceneVersion);
```

Comment:
Should it return the created version or the upgraded version (if it exists) instead of Version.CURRENT?